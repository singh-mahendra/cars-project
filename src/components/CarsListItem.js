import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

const CarsListItem = (props) =>{
    return (
        <section className="car-list-item">
            <aside>
                <img src={props.pictureUrl} width="100" height="100"></img>
            </aside>
            <content>
                <div>
                    <h4>{props.manufacturerName} - {props.modelName}</h4>
                </div>
                <div>
                    <span>Stock # {props.stockNumber} - {props.mileage.number} {props.mileage.unit} - {props.fuelType} - {props.color}</span>
                </div>
                <Link to="/detail" onClick={props.onClick} id={props.id}>More Details</Link>
            </content>
        </section>
    )
}

CarsListItem.propTypes = {
    manufacturerName: PropTypes.string,
    id: PropTypes.number,
    onClick: PropTypes.func
}

export default CarsListItem;