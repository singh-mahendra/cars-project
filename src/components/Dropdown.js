import React, {useState, useCallback} from 'react'
import PropTypes from 'prop-types';

const Dropdown = (props) => {
    const [selectedKey, updateSelectedKey] = useState("None");
    const setSelectedKey = useCallback((event) => {
        const value = event.target.getAttribute("data-value");
        const displayValue = event.target.getAttribute("display-value");
        const updatedValue = displayValue ? displayValue : value;
        updateSelectedKey(updatedValue);
        props.onSelect(value);
    }, []);


    return(
        <div className="dropdown">
            <button className="dropbtn">{selectedKey}</button>
            <div className="dropdown-content" onClick={setSelectedKey}>
                {
                    props.items.map((item) => {
                        return props.itemComponent(item);
                    })
                }
            </div>
        </div>
    )
}

Dropdown.propTypes = {
    items: PropTypes.array,
    onSelect: PropTypes.func,
    itemComponent: PropTypes.func
}

export default Dropdown;