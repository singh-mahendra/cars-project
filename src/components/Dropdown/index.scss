$color-white:#EDEDED;
$color-black:#4A4A4A;
$color-orange:#EA7F28;
$color-brown:#D37324;
$bg-white:#FFF;
$buttonheight: 32px;
$buttonwidth:128px;

.dropbtn {
	background-color: white;
	color: black;
	padding: 8px;
	font-size: 16px;
	border: none;
	width: 20rem;
	border: 1px solid lightgray;
	border-radius: 5px;
	text-align: left;
	background: url("../../images/down-arrow.png") no-repeat scroll center center / 100% auto rgba(0, 0, 0, 0);
	background-size: 24px;
	background-position: right;
	text-transform: capitalize;
  }
  
  .dropdown-content.visible{
	display: block;
  }
  
  .dropdown-content.hidden{
	display: none;
  }
  
  .dropdown {
	position: relative;
	display: inline-block;
	padding: 0.2rem;
  }
  
  .dropdown-content {
	position: absolute;
	background-color: #FFF;
	min-width: 20rem;
	box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
	z-index: 1;
	a {
		color: black;
		padding: 8px;
		text-decoration: none;
		display: block;
		text-transform: capitalize;
		text-align: left;
	  }
	a:hover {
		background-color: $color-orange;
	}
	.dropdown-content {
		border-radius: 5px;
	}
	.dropbtn {
		display: block;
		background-color: $color-orange;
	}
  }
  