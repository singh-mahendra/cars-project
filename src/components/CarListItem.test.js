import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from './Dropdown';
import CarsListItem from './CarsListItem';

describe("Carlistitem", () => {
  it("should render my component", () => {
    const id = 1;
    const wrapper = shallow(<CarsListItem id={id} manufacturerName="Tesla" pictureUrl="#" modelName="X" mileage={{unit: "KM", number:"1"}}/>);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});