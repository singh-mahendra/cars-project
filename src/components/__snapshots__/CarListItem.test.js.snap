// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carlistitem should render my component 1`] = `
Array [
  <section
    className="car-list-item"
  >
    <aside>
      <img
        height="100"
        src="#"
        width="100"
      />
    </aside>
    <content>
      <div>
        <h3
          className="car-list-item-header"
        >
          Tesla
           - 
          X
        </h3>
      </div>
      <div
        className="car-list-detail"
      >
        <span>
          Stock # 
           - 
          1
           
          KM
           - 
           - 
        </span>
      </div>
      <Link
        id={1}
        to="/detail"
      >
        More Details
      </Link>
    </content>
  </section>,
]
`;
